homeassistant/helpers/network.py

```diff
@@ -171,6 +184,32 @@ def _get_request_host() -> str | None:
     return yarl.URL(request.url).host
 
 
+@bind_hass
+def _get_hostname_url(
+    hass: HomeAssistant,
+    *,
+    require_current_request: bool = False,
+    require_ssl: bool = False,
+    require_standard_port: bool = False,
+) -> str:
+    """Get hostname of this instance."""
```

Comment:
I think we should change the name of this function and/or docstring to reflect that it requires the Supervisor.